graylog-provider/src/main/java/com/busybusy/graylog_provider/EventJsonizer.java

```diff
@@ -0,0 +1,290 @@
+        sb.append("\"host\": \"").append(HOST).append("\", ");
+        sb.append("\"short_message\": \"").append(getSafeSizeString(event.name())).append("\", ");
+
+        if (attributes.contains("full_message"))
+        {
+            String fullMessage = eventAttributes.get("full_message").toString();
+            sb.append("\"full_message\": \"").append(getSafeSizeString(fullMessage)).append("\", ");
+        }
+        // No else needed: this is an optional long message (may contain a backtrace)
+
+        if (attributes.contains("timestamp"))
+        {
+            String timestamp = eventAttributes.get("timestamp").toString();
+            sb.append("\"timestamp\": ").append(getSafeSizeString(timestamp)).append(", ");  // user-provided timestamp
+        }
+        else
+        {
+            sb.append("\"timestamp\": ").append(System.currentTimeMillis() / 1000d).append(", ");
+        }
+
+        if (attributes.contains("level"))
+        {
+            String level = eventAttributes.get("level").toString();
+            sb.append("\"level\": ").append(getSafeSizeString(level)); // user-provided syslog level
+        }
+        else
+        {
+            sb.append("\"level\": 6"); // Informational
+        }
+
+        for (String attribute : attributes)
+        {
+            //noinspection StatementWithEmptyBody
+            if (attribute.equalsIgnoreCase("version") ||
+                    attribute.equalsIgnoreCase("host") ||
+                    attribute.equalsIgnoreCase("short_message") ||
+                    attribute.equalsIgnoreCase("full_message") ||
+                    attribute.equalsIgnoreCase("timestamp") ||
+                    attribute.equalsIgnoreCase("level"))
+            {
+                // do nothing, these fields have already been handled above
+            }
+            else
+            {
+                String jsonAttribute = attribute.replaceAll("\\s", "_"); // Graylog omits fields whose name have white space, replace with '_'
+                sb.append(underscorePrefix(jsonAttribute));
+
+                Object attributeValue = eventAttributes.get(attribute);
+                if (attributeValue instanceof String) // gotta use escape quotes for JSON strings
+                {
+                    sb.append(jsonAttribute).append("\": \"").append(getSafeSizeString((String) attributeValue)).append("\"");
+                }
+                else if (attributeValue instanceof List)
+                {
+                    sb.append(jsonAttribute).append("\": ").append(getJsonFromListRecursive((List) attributeValue));
+                }
+                else if (attributeValue instanceof Map)
+                {
+                    //noinspection unchecked
+                    sb.append(jsonAttribute).append("\": ").append(getJsonFromMapRecursive((Map<String, Object>) attributeValue));
+                }
+                else if (attributeValue instanceof Integer ||
```

Comment:
seems like you could have an isNumber() method that would return a boolean and execute this OR statement and use it in the 3 spots that do this check